### PR TITLE
Automatically generate Weave documentation

### PR DESCRIPTION
**Pull request by davidwallacejackson**
Goes with changes in the private wandb/core repo to automatically generate documentation based on docstrings for Weave ops.

This new version will:

* check out core
* install node and the dependencies for core's `cg` library
* run a node-based documentation generator script in cg, which outputs a number of markdown files
* copy those markdown files into the docugen repo
* merge the core-generated markdown into the other markdown files created by docugen

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -36,13 +36,38 @@ runs:
...
+      run: |
+        yarn --cwd=./repos/core/lib/js/cg --frozen-lockfile
+        yarn --cwd=./repos/core/lib/js/cg generate-docs
+        rm -rf ./repos/gitbook/ref/weave/**
+        mkdir -p ./repos/gitbook/ref/weave
+        cp -r ./repos/core/lib/js/cg/docs_gen/* ./repos/gitbook/ref/weave
+
     # generate the docs from the latest client library and overwrite gitbook contents
-    - name: generate documentation
+    - name: generate client documentation
```
Comment (dmitryduev) [inline action.yml:64]: 😸 ?
```suggestion
    - name: generate SDK documentation
```

File: action.yml [hunk 2]
```diff
@@ -62,6 +87,7 @@ runs:
           git add -A .
           git commit -m "update reference docs to cli@${{ inputs.client-branch }}, core@${{ inputs.core-branch }}"
           git remote set-url origin https://${{ inputs.access-token }}@github.com/wandb/gitbook.git ; \
+          git pull ; \
```
Comment (dmitryduev) [inline action.yml:90]: Why is this necessary now?

**Review by dmitryduev — COMMENTED** [review]
Thanks @davidwallacejackson. I really like the idea to generate all docs from the same GHA, however it does feel a bit weird to have it in multiple repositories + we would have to keep all "copies" up to date. 
What do think about the following scheme:
- Set up a GH action in client and core that would send a [repository dispatch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) event to this repo on merges to master.
- Run this action here with `on: repository_dispatch`.
- Profit?

Also, how long does it take to run the whole thing, currently?

File: action.yml [hunk 3]
```diff
@@ -36,13 +36,38 @@ runs:
         VERSION: latest
       run: python -m pip install -r ${{ github.action_path }}/requirements.txt git+https://github.com/wandb/client.git@$VERSION
```
Comment (dmitryduev) [inline action.yml:37]: BTW, what is `$VERSION` here? It should be pointed at the latest release (and not master), right?

Comment (davidwallacejackson) [reply action.yml:37]: yep! on line 36 it's set to `latest`

File: action.yml (hunk 2 above)
Comment (davidwallacejackson) [reply action.yml:90]: in case new changes have been made from another source, since there's gonna be two different projects updating the docs now

Comment (dmitryduev) [reply action.yml:90]: as in "while this action has been running in the current repo"? yea, makes sense, thnx.

**Comment by davidwallacejackson** [comment]
It currently takes about a minute to run the whole thing. I think the repository dispatch idea is probably safer than what I did, but not by so much that I'd want to hold this up, if that's OK.

**Review by dmitryduev — APPROVED** [review]
